rs-client-demo/src/main.rs

```diff
@@ -22,71 +24,92 @@ pub struct LatestChangeEntry {
+    } else {
+        // The `product-integrity` collection is automatically created in CI because
+        // we use the `testing.ini` config which will create all signed resources on startup.
+        // See `KINTO_SIGNER_RESOURCES` in `.circleci/config.yml`.
+        let collection = option_env!("COLLECTION").unwrap_or("product-integrity");
+
+        println!("Connect to local server {}", server_url);
+        let editor_client = Client::builder()
+            .http_client(Box::new(ViaductClient))
+            .server_url(server_url)
+            .authorization(authorization)
+            .collection_name(collection)
+            .build()
+            .unwrap();
+
+        println!("main-workspace/{}: Create a record", collection);
+        editor_client
+            .store_record(Record::new(json!({
+            "id": "my-key",
+            "foo": "bar"
+            })))
+            .await?;
+
+        println!("main-workspace/{}: Request review from peers", collection);
+        editor_client.request_review("I made changes").await?;
+
+        print!(
+            "main-workspace/{}: Fetching preview records with default `Verifier`...",
+            collection
+        );
+        let mut preview_client = Client::builder()
+            .http_client(Box::new(ViaductClient))
+            .server_url(server_url)
+            .bucket_name("main-preview")
+            .collection_name(collection)
+            .build()
+            .unwrap();
 
-    let records = preview_client.get().await?;
-    println!("{} record(s) published", records.len());
+        let records = preview_client.get().await?;
+        println!("{} record(s) published", records.len());
+    }
 
     println!("\n\n");
 
-    println!("Fetch all Remote Settings collections from PROD server.");
-    let url = "https://firefox.settings.services.mozilla.com/v1/buckets/monitor/collections/changes/changeset?_expected=0";
-    let response =
-        tokio::task::spawn_blocking(move || Request::get(Url::parse(url).unwrap()).send().unwrap())
-            .await
-            .unwrap();
+    let server_url = match env_name.as_str() {
+        "dev" => "https://remote-settings-dev.allizom.org/v1",
+        "stage" => "https://remote-settings.allizom.org/v1",
+        _ => DEFAULT_SERVER_URL,
+    };
+    let root_hash = match env_name.as_str() {
+        "dev" => "45:C3:7F:3A:09:A6:D7:0E:0F:A3:21:FB:29:75:3B:A7:99:8F:12:59:B3:27:72:76:8F:23:CC:DC:24:83:67:98",
+        "stage" => "3C:01:44:6A:BE:90:36:CE:A9:A0:9A:CA:A3:A5:20:AC:62:8F:20:A7:AE:32:CE:86:1C:B2:EF:B7:0F:A0:C7:45",
+        _ => PROD_CERT_ROOT_HASH,
+    };
+
+    println!(
+        "Fetch all Remote Settings collections from {} server.",
+        env_name.to_uppercase()
+    );
+    let poll_url = format!(
+        "{}/buckets/monitor/collections/changes/changeset?_expected=0",
+        server_url
+    );
+    let response = tokio::task::spawn_blocking(move || {
```

Comment:
I feel like we can make this easier to read. But that might require adding a higher level http client like reqwest.